Userland/Libraries/LibTimeZone/TimeZone.cpp

```diff
@@ -82,7 +85,33 @@ StringView system_time_zone()
 
 StringView current_time_zone()
 {
-    return canonicalize_time_zone({ tzname[0], __builtin_strlen(tzname[0]) }).value_or("UTC"sv);
+    if (char* tz = getenv("TZ"); tz != nullptr) {
+        // FIXME: Actually parse the TZ environment variable, described here:
+        // https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08
+        StringView time_zone { tz, strlen(tz) };
+
+        if (auto maybe_time_zone = canonicalize_time_zone(time_zone); maybe_time_zone.has_value())
+            return *maybe_time_zone;
+    }
+
+#if __serenity__
+    return system_time_zone();
+#else
+    static constexpr auto zoneinfo = "/zoneinfo/"sv;
+    char buffer[PATH_MAX];
+
+    if (auto size = readlink("/etc/localtime", buffer, sizeof(buffer)); size > 0) {
+        StringView time_zone { buffer, static_cast<size_t>(size) };
+
+        if (auto index = time_zone.find(zoneinfo); index.has_value())
+            time_zone = time_zone.substring_view(*index + zoneinfo.length());
+
+        if (auto maybe_time_zone = canonicalize_time_zone(time_zone); maybe_time_zone.has_value())
+            return *maybe_time_zone;
+    }
+
+    return "UTC"sv;
```

Comment:
Maybe we should put a log here?
Since it would help tracking similar issues?

(If this is a very common case than perhaps we shouldn't)

Reply:
It doesn't get hit on my system, but I added logging here behind a debug flag